include/pybind11/functional.h

```diff
@@ -75,6 +84,36 @@ struct type_caster<std::function<Return(Args...)>> {
             // PYPY segfaults here when passing builtin function like sum.
             // Raising an fail exception here works to prevent the segfault, but only on gcc.
             // See PR #1413 for full details
+        } else {
+            // Check number of arguments of Python function
+            auto getArgCount = [&](PyObject *obj) {
+                // This is faster then doing import inspect and inspect.signature(obj).parameters
+                auto *t = PyObject_GetAttrString(obj, "__code__");
+                auto *argCount = PyObject_GetAttrString(t, "co_argcount");
+                return PyLong_AsLong(argCount);
+            };
+            long argCount = -1;
+
+            if (static_cast<bool>(PyObject_HasAttrString(src.ptr(), "__code__"))) {
```

Comment:
static_cast<bool> makes this code look more confusing than it should be. Either just have an if statement directly on HasAttrString or explicitly do if (has_attr_string == 1)

Reply:
I'd try to only use `PyObject_GetAttrString()` here and not `PyObject_HasAttrString()` at all (unless you really never actually need the attribute).

I looked at this in depth in some other context (a couple years ago probably, but I'd bet it's still the same) and found that `PyObject_HasAttrString()` uses `PyObject_GetAttrString()` and just throws the object away.

Reply:
General: when you feel you're done, I'd review this code do see what can be factored out to a non-templated `inline` function (or functions). That might help the compiler a little, but is also nice for humans reading the code.

Reply:
Ok I used `PyObject_GetAttrString` and `.attr`, where I'm sure that the attribute is existing, since there I didn't want to write `try ... catch` around `.attr` for performance reasons.

Reply:
Ok I found out how to use `getattr`. I hope this is the way to go.